marimo/_runtime/dataflow/edges.py

```diff
@@ -0,0 +1,269 @@
+# Copyright 2024 Marimo. All rights reserved.
+"""Edge computation logic for cell dependencies."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Literal, Optional, cast
+
+from marimo import _loggers
+
+if TYPE_CHECKING:
+    from marimo._ast.cell import CellImpl
+    from marimo._ast.sql_visitor import SQLRef
+    from marimo._ast.visitor import Name
+    from marimo._runtime.dataflow.definitions import DefinitionRegistry
+    from marimo._runtime.dataflow.topology import GraphTopology
+    from marimo._types.ids import CellId_t
+
+from marimo._ast.sql_visitor import SQLTypes
+
+LOGGER = _loggers.marimo_logger()
+
+
+class EdgeComputer:
```

Comment:
In implementation used as basically a namespace. I think it's worth just exposing as just functions opposed to a class

Reply:
the refactor made the writing of the class args a bit more verbose (passing them all the way through multiple levels), so i plan to leave this as-is, but i can revisit